cpp/src/arrow/acero/aggregate_node.cc

```diff
@@ -602,7 +621,7 @@ class GroupByNode : public ExecNode, public TracedNode {
     for (const auto& segment_key_field_id : segment_key_field_ids) {
       if (key_field_id_set.find(segment_key_field_id) != key_field_id_set.end()) {
         return Status::Invalid("Group-by aggregation with field '",
-                               input_schema->field(segment_key_field_id)->name(),
```

Comment:
Any reason that we need to change `input_schema` from a pointer reference to a schema reference?

Reply:
I think this can be a `shared_ptr`, if you prefer.

Reply:
Yeah I think `const shared_ptr<Schema>&` is better because then some of the lines using input_schema can be unchanged.

Reply:
Fixed.